libs/isograph-react/src/core/read.ts

```diff
@@ -503,7 +504,11 @@ function readData<TReadFromStore>(
                   const [networkRequest, disposeNetworkRequest] =
                     maybeMakeNetworkRequest(
                       environment,
-                      entrypoint,
+                      entrypoint as IsographEntrypoint<
```

Comment:
LGTM overall. It might make sense to try to avoid this by not having `IsographEntrypoint<T1, T2>` but having `IsographEntrypoint<T1, T2, NormalizationAst>` in various places, or changing the type of `maybeMakeNetworkRequest`.

I don't know if that's possible.